pulsar-common/src/main/java/com/yahoo/pulsar/common/api/Commands.java

```diff
@@ -611,7 +611,6 @@ public static long initBatchMessageMetadata(PulsarApi.MessageMetadata.Builder me
         messageMetadata.setPublishTime(builder.getPublishTime());
         messageMetadata.setProducerName(builder.getProducerName());
         messageMetadata.setSequenceId(builder.getSequenceId());
-        messageMetadata.setReplicatedFrom(builder.getReplicatedFrom());
```

Comment:
Can you also add a test in the replication unit test for batch messages? It should have caught this